File: ex1functions.m
% ex1: Functions, MATLAB

gcd = euclidgcd(252, 105);
disp(gcd);


function foo = euclidgcd(foo, bar)
    % EUCLIDGCD Return greatest common divisor

    while foo ~= bar

        if foo > bar
            foo = foo - bar;
        else
            bar = bar - foo;
        end
    end
end

File: ex1functions.py
# ex1: Functions, Python

def euclid_gcd(a, b):
    """Return greatest common divisor."""

    while a != b:

        if a > b:
            a = a - b
        else:
            b = b - a

    return a


gcd = euclid_gcd(252, 105)
print(gcd)